server/fleet/datastore.go

```diff
@@ -605,7 +606,7 @@ type Datastore interface {
 	//
 	// After aggregation, it cleans up unused software (e.g. software installed
 	// on removed hosts, software uninstalled on hosts, etc.)
-	SyncHostsSoftware(ctx context.Context, updatedAt time.Time) error
+	SyncHostsSoftware(ctx context.Context, updatedAt time.Time, logger log.Logger) error
```

Comment:
Important change 3 (which is why everything else had to change)

Reply:
tbh didn't realize we even had a logger in the dsta store, but also yeah we'd be missing the cron logger context so it wouldn't be clear that this is part of the vulns cron.

Reply:
Yeah I was gonna recommend using the `ds.logger`, but I traced the code and this makes more sense. 

I think we should refactor (not in this PR ofc) the `SyncHostsSoftware` function to not be a datastore method and instead just be a "service" function that takes its dependencies (like the cron logger) as arguments.